apps/dashboard/lib/auth/workos.ts

```diff
@@ -0,0 +1,662 @@
+import { WorkOS } from "@workos-inc/node";
+import { OAuthResult, Organization, OrgMembership, Membership, UNKEY_SESSION_COOKIE, User, type SignInViaOAuthOptions, UpdateOrgParams, OrgInvite, Invitation, OrgInvitation, UpdateMembershipParams, SessionValidationResult, SessionData, AuthProviderError, AuthErrorCode } from "./types";
+import { env } from "@/lib/env";
+import { getCookie, updateCookie } from "./cookies";
+import { BaseAuthProvider } from "./base-provider"
+
+const SIGN_IN_REDIRECT = "/apis";
+const SIGN_IN_URL = "/auth/sign-in";
+
+export class WorkOSAuthProvider extends BaseAuthProvider {
+  private static instance: WorkOSAuthProvider | null = null;
+  private static provider: WorkOS;
+  private static clientId: string;
+
+  constructor(config: { apiKey: string; clientId: string }) {
+    super();
+    if (WorkOSAuthProvider.instance) {
+      return WorkOSAuthProvider.instance;
+    }
+
+    WorkOSAuthProvider.clientId = config.clientId;
+    WorkOSAuthProvider.provider = new WorkOS(config.apiKey, { clientId: config.clientId });
+    WorkOSAuthProvider.instance = this;
+  }
+
+  async validateSession(sessionToken: string): Promise<SessionValidationResult> {
+    if (!sessionToken) {
+      return { isValid: false, shouldRefresh: false };
+    }
+
+    const WORKOS_COOKIE_PASSWORD = env().WORKOS_COOKIE_PASSWORD;
```

Comment:
it might be easier if we store this in the class itself and fail during the constructor, rather than checking in every method individually